v2/tools/asoctl/internal/importing/arm_resource_importer.go

```diff
@@ -0,0 +1,235 @@
+		return nil, err // arm.ParseResourceID already returns a good error, no need to wrap
+	}
+
+	// Create a blank object into which we capture the current state of the resource
+	obj, err := ri.createBlankObjectFromID(id)
+	if err != nil {
+		// Error doesn't need additional context
+		return nil, err
+	}
+
+	importable, ok := obj.(genruntime.ImportableARMResource)
+	if !ok {
+		return nil, errors.Errorf(
+			"unable to create blank resource, expected %s to identify an importable ARM object", armID)
+	}
+
+	status, err := ri.getStatus(ctx, armID, importable)
+	if err != nil {
+		return nil, err
+	}
+
+	err = importable.InitializeSpec(status)
+	if err != nil {
+		return nil, errors.Wrapf(err, "setting status on Kubernetes resource for resource %s", armID)
+	}
+
+	return &ResourceImportResult{
+		resources: []genruntime.MetaObject{importable},
+	}, nil
+}
+
+// getStatus gets the status of the resource with the specified ID from ARM
+func (ri *ARMResourceImporter) getStatus(ctx context.Context, armID string, armMeta genruntime.ARMMetaObject) (genruntime.ConvertibleStatus, error) {
+	// Create a request to get the current state of the resource
+	req, err := ri.createRequest(ctx, armID, armMeta.GetAPIVersion())
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create request to import ARM resource %s", armID)
+	}
+
+	// Execute the request
+	resp, err := ri.client.Do(req)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to execute request to import ARM resource %s", armID)
+	}
+
+	if !azruntime.HasStatusCode(resp, http.StatusOK) {
+		return nil, azruntime.NewResponseError(resp)
+	}
+
+	klog.V(3).Infof("Request succeeded")
+
+	armStatus, err := genruntime.NewEmptyARMStatus(armMeta, ri.Scheme())
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create empty ARM status for importing ARM resource %s", armID)
+	}
+
+	// Populate our Status from the response
+	if err = azruntime.UnmarshalAsJSON(resp, armStatus); err != nil {
+		return nil, errors.Wrapf(err, "unable to deserialize ARM response for importing ARM resource %s", armID)
+	}
+
+	// Convert the ARM shape to the Kube shape
+	status, err := genruntime.NewEmptyVersionedStatus(armMeta, ri.Scheme())
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to construct status object for resource: %s", armID)
+	}
+
+	//TODO: what if it's not ok
+	if s, ok := status.(genruntime.FromARMConverter); ok {
+		var knownOwner genruntime.ArbitraryOwnerReference
+		err = s.PopulateFromARM(knownOwner, reflecthelpers.ValueOfPtr(armStatus)) // TODO: PopulateFromArm expects a value... ick
+		if err != nil {
+			return nil, errors.Wrapf(err, "converting ARM status to Kubernetes status for resource %s", armID)
+		}
+	}
+
+	return status, nil
+}
+
+// createRequest constructs the request to GET the ARM resource
+func (ri *ARMResourceImporter) createRequest(ctx context.Context, armID string, apiVersion string) (*policy.Request, error) {
```

Comment:
I don't think you should need to do this here. We have `v2/internal/genericarmclient/generic_client.go` which has `func (client *GenericClient) GetByID(ctx context.Context, resourceID string, apiVersion string, resource interface{}) (time.Duration, error)` - you can use that instead of this I believe.

Reply:
Now using `genericarmclient`.